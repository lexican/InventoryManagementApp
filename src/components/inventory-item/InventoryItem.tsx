import {useNavigation} from '@react-navigation/native';
import React, {FC} from 'react';
import {IInventory} from '../../types/types';
import {
  EditInventoryBtn,
  EditInventoryBtnText,
  InventoryItemContainer,
  InventoryName,
  InventoryPrice,
  InventoryTotalStock,
} from './InventoryItem.styles';

type IProps = {
  inventory: IInventory;
};

type Nav = {
  navigate: (value: string, data: any) => void;
};

const InventoryItem: FC<IProps> = ({inventory}) => {
  const {navigate} = useNavigation<Nav>();
  return (
    <InventoryItemContainer>
      <InventoryName>{inventory.name}</InventoryName>
      <InventoryPrice>#{inventory.price}</InventoryPrice>
      <InventoryTotalStock>{inventory.totalStock}</InventoryTotalStock>
      <EditInventoryBtn
        testID="edit"
        activeOpacity={0.9}
        onPress={() => {
          navigate('CreateInventory', {
            inventoryItem: inventory,
          });
        }}>
        <EditInventoryBtnText>Edit</EditInventoryBtnText>
      </EditInventoryBtn>
    </InventoryItemContainer>
  );
};

export default InventoryItem;
