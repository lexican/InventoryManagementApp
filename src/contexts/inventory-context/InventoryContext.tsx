import React, {
  createContext,
  FC,
  useContext,
  useEffect,
  useMemo,
  useState,
} from 'react';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {IInventory} from '../../types/types';
import {AsyncStorageInventoriesKey} from '../../utils/constants';

export const InventoryStateContext = createContext<IStateContext>(
  undefined as never,
);

type IStateContext = {
  inventories: IInventory[];
  addInventory: (inventory: IInventory) => void;
  updateInventory: (inventory: IInventory, name: string) => void;
  deleteInventory: (name: string) => void;
  clearInventories: () => void;
};

type IProp = {
  children: JSX.Element;
};

export const InventoryContextProvider: FC<IProp> = ({children}) => {
  const [inventories, setInventories] = useState<IInventory[]>([]);

  useEffect(() => {
    getInventories();
  }, []);

  const getInventories = async () => {
    const inventoryItems = await AsyncStorage.getItem(
      AsyncStorageInventoriesKey,
    );
    if (inventoryItems != null) {
      setInventories(JSON.parse(inventoryItems));
    }
  };

  const addInventory = async (inventory: IInventory) => {
    setInventories([...inventories, inventory]);
    await updateAsyncStorageData([...inventories, inventory]);
  };

  const updateInventory = async (inventory: IInventory, name: string) => {
    const tempInventories = [...inventories];

    const inventoryIndex = tempInventories.findIndex(
      inventoryItem => inventoryItem.name.toLowerCase() === name.toLowerCase(),
    );

    if (inventoryIndex !== -1) {
      tempInventories[inventoryIndex] = {...inventory};
      setInventories([...tempInventories]);
      await updateAsyncStorageData(tempInventories);
    }
  };

  const deleteInventory = async (name: string) => {
    const tempInventories = inventories.filter(
      inventory => inventory.name.toLowerCase() !== name.toLowerCase(),
    );
    setInventories([...tempInventories]);
    await updateAsyncStorageData(tempInventories);
  };

  const clearInventories = async () => {
    await AsyncStorage.removeItem(AsyncStorageInventoriesKey);
    setInventories([]);
  };

  const updateAsyncStorageData = async (inventories: IInventory[]) => {
    await AsyncStorage.setItem(
      AsyncStorageInventoriesKey,
      JSON.stringify(inventories),
    );
  };

  const value = useMemo(
    () => ({
      inventories,
      addInventory,
      updateInventory,
      deleteInventory,
      clearInventories,
    }),
    [
      inventories,
      addInventory,
      updateInventory,
      deleteInventory,
      clearInventories,
    ],
  );

  return (
    <InventoryStateContext.Provider value={value}>
      {children}
    </InventoryStateContext.Provider>
  );
};

export function useInventoryStateContext(): IStateContext {
  return useContext(InventoryStateContext);
}
