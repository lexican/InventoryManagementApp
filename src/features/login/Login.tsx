import React, {useState} from 'react';
import CustomTextInput from '../../components/custom-text-input/CustomTextInput';
import {
  LoginBtn,
  LoginBtnText,
  LoginContainer,
} from './Login.styles';

const Login = () => {
  const [email, setEmail] = useState<string>('');

  const [password, setPassword] = useState<string>('');

  const handleLogin = () => {};

  return (
    <LoginContainer>
      <CustomTextInput
        placeholder="Email"
        value={email}
        onChangeText={email => {
          setEmail(email);
        }}
      />
      <CustomTextInput
        placeholder="Password"
        secureTextEntry={true}
        value={password}
        onChangeText={password => {
          setPassword(password);
        }}
      />
      <LoginBtn activeOpacity={0.9} onPress={handleLogin}>
        <LoginBtnText>Login</LoginBtnText>
      </LoginBtn>
    </LoginContainer>
  );
};

export default Login;
