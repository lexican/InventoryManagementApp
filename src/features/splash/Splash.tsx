import React, {useEffect} from 'react';
import {SplashContainer, SplashTitle} from './Splash.styles';

type IProps = {};

export default function Splash({}: IProps): JSX.Element {
  useEffect(() => {
    const timeout = setTimeout(() => {
      //Navigate to home or Login after 3 seconds
    }, 3000);
    return () => clearTimeout(timeout);
  }, []);
  return (
    <SplashContainer>
      <SplashTitle>Inventory</SplashTitle>
    </SplashContainer>
  );
}
