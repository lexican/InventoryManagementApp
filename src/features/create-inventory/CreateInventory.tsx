import {RouteProp, useRoute} from '@react-navigation/native';
import React, {FC, useState} from 'react';
import CustomTextInput from '../../components/custom-text-input/CustomTextInput';
import {useInventoryStateContext} from '../../contexts/inventory-context/InventoryContext';
import {showToast} from '../../utils/utils';
import {
  CreateInventoryBtn,
  CreateInventoryBtnText,
  CreateInventoryErrorText,
  CreateInventorySafeAreaView,
  CreateInventoryScrollView,
  CreateInventoryText,
  DeleteInventoryBtn,
  DeleteInventoryBtnText,
  HeaderBackBtn,
  HeaderBackBtnText,
  HeaderRow,
} from './CreateInventory.styles';

type IProp = {
  route: any;
  navigation: any;
};

type ParamList = {
  CreateInventory: {
    inventoryItem: any;
  };
};

const CreateInventory: FC<IProp> = ({navigation}) => {
  const [isDeleteInventoryModalVisible, setDeleteInventoryModalVisible] =
    useState<boolean>(false);
  const route = useRoute<RouteProp<ParamList, 'CreateInventory'>>();
  const {inventoryItem: editInventoryItem} = route.params;
  const [inventoryName, setInventoryName] = useState<string>(
    editInventoryItem?.name ?? '',
  );
  const [inventoryTotalStock, setInventoryTotalStock] = useState<string>(
    editInventoryItem?.totalStock?.toString() ?? '',
  );
  const [inventoryPrice, setInventoryPrice] = useState<string>(
    editInventoryItem?.price?.toString() ?? '',
  );
  const [inventoryDescription, setInventoryDescription] = useState<string>(
    editInventoryItem?.description ?? '',
  );

  const [inventoryNameError, setInventoryNameError] = useState<string>('');
  const [inventoryTotalStockError, setInventoryTotalStockError] =
    useState<string>('');
  const [inventoryPriceError, setInventoryPriceError] = useState<string>('');
  const [inventoryDescriptionError, setInventoryDescriptionError] =
    useState<string>('');

  const {addInventory, inventories, updateInventory, deleteInventory} =
    useInventoryStateContext();

  const createInventory = () => {
    setInventoryNameError('');
    setInventoryTotalStockError('');
    setInventoryPriceError('');
    setInventoryDescriptionError('');
    if (inventoryName === '') {
      setInventoryNameError('Inventory Name is Required');
    } else if (inventoryTotalStock === '') {
      setInventoryTotalStockError('Inventory Stock is Required');
    } else if (inventoryPrice === '') {
      setInventoryPriceError('Inventory Price is Required');
    } else if (inventoryDescription === '') {
      setInventoryDescriptionError('Inventory Description is Required');
    } else if (inventoryDescription !== '' && inventoryDescription.length < 3) {
      setInventoryDescriptionError(
        'Inventory Description must be at least 3 characters',
      );
    } else {
      if (editInventoryItem?.name == null) {
        // Create New Inventory
        const inventory = inventories.find(
          inventoryItem =>
            inventoryItem.name.toLowerCase() === inventoryName.toLowerCase(),
        );
        if (inventory != null) {
          setInventoryNameError('Inventory Name already exist');
        } else {
          addInventory({
            name: inventoryName,
            description: inventoryDescription,
            price: Number(inventoryPrice),
            totalStock: Number(inventoryTotalStock),
          });
          setInventoryName('');
          setInventoryTotalStock('');
          setInventoryPrice('');
          setInventoryDescription('');
        }

        showToast('Inventory Created.');
      } else {
        //Update Inventory
        if (
          inventoryName.toLowerCase() ===
          editInventoryItem?.name.toLowerCase().toString()
        ) {
          updateInventory(
            {
              name: inventoryName,
              description: inventoryDescription,
              price: Number(inventoryPrice),
              totalStock: Number(inventoryTotalStock),
            },
            editInventoryItem?.name,
          );
          showToast('Inventory Updated');
        } else {
          const inventory = inventories.find(
            inventoryItem =>
              inventoryItem.name.toLowerCase() === inventoryName.toLowerCase(),
          );
          if (inventory != null) {
            setInventoryNameError('Inventory Name already exist');
          } else {
            updateInventory(
              {
                name: inventoryName,
                description: inventoryDescription,
                price: Number(inventoryPrice),
                totalStock: Number(inventoryTotalStock),
              },
              editInventoryItem?.name,
            );
            showToast(`Inventory Updated`);
          }
        }
      }
    }
  };

  const goBack = () => {
    navigation.goBack();
  };

  return (
    <CreateInventorySafeAreaView>
      <CreateInventoryScrollView>
        <HeaderRow>
          <HeaderBackBtn onPress={goBack} activeOpacity={0.9}>
            <HeaderBackBtnText>&#60;</HeaderBackBtnText>
          </HeaderBackBtn>
          <CreateInventoryText>
            {editInventoryItem?.name == null ? 'Create' : 'Edit'} Inventory
          </CreateInventoryText>
        </HeaderRow>

        <CustomTextInput
          placeholder="Inventory Name"
          value={inventoryName}
          onChangeText={name => {
            setInventoryName(name);
          }}
        />
        {inventoryNameError && (
          <CreateInventoryErrorText>
            {inventoryNameError}
          </CreateInventoryErrorText>
        )}
        <CustomTextInput
          placeholder="Inventory Total Stock"
          value={inventoryTotalStock}
          onChangeText={totalStock => {
            setInventoryTotalStock(totalStock.replace(/[^0-9]/g, ''));
          }}
          keyboard="numeric"
        />
        {inventoryTotalStockError && (
          <CreateInventoryErrorText>
            {inventoryTotalStockError}
          </CreateInventoryErrorText>
        )}
        <CustomTextInput
          placeholder="Inventory Price"
          value={inventoryPrice}
          onChangeText={price => {
            setInventoryPrice(price);
          }}
          keyboard="numeric"
        />
        {inventoryPriceError && (
          <CreateInventoryErrorText>
            {inventoryPriceError}
          </CreateInventoryErrorText>
        )}
        <CustomTextInput
          placeholder="Inventory Description"
          value={inventoryDescription}
          onChangeText={description => {
            setInventoryDescription(description);
          }}
        />
        {inventoryDescriptionError && (
          <CreateInventoryErrorText>
            {inventoryDescriptionError}
          </CreateInventoryErrorText>
        )}
        <CreateInventoryBtn
          activeOpacity={0.9}
          onPress={() => {
            createInventory();
          }}>
          <CreateInventoryBtnText>
            {editInventoryItem?.name == null ? 'Create' : 'Edit'}
          </CreateInventoryBtnText>
        </CreateInventoryBtn>
        {editInventoryItem?.name != null && (
          <DeleteInventoryBtn
            activeOpacity={0.9}
            onPress={() => {
              setDeleteInventoryModalVisible(true);
            }}>
            <DeleteInventoryBtnText>Delete</DeleteInventoryBtnText>
          </DeleteInventoryBtn>
        )}
      </CreateInventoryScrollView>
    </CreateInventorySafeAreaView>
  );
};

export default CreateInventory;
