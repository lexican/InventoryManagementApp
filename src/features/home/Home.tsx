import {NativeStackScreenProps} from '@react-navigation/native-stack';
import React, {useState} from 'react';
import ConfirmModal from '../../components/confirm-modal/ConfirmModal';
import InventoryList from '../../components/inventory-list/InventoryList';
import {useAuthStateContext} from '../../contexts/auth-context/AuthContext';
import {useInventoryStateContext} from '../../contexts/inventory-context/InventoryContext';
import {RootStackParamList} from '../../navigation/navigation';
import {
  AddInventoryBtn,
  AddInventoryBtnText,
  FlexRow,
  HomeContainer,
  LogoutBtn,
  LogoutBtnText,
  LogoutButtonContainer,
} from './Home.styles';

type Prop = NativeStackScreenProps<RootStackParamList, 'CreateInventory'>;

const Home = ({navigation}: Prop) => {
  const [isLogoutInventoryModalVisible, setLogoutInventoryModalVisible] =
    useState(false);
  const navigateToCreateInventory = () => {
    navigation.navigate('CreateInventory', {inventoryItem: {}});
  };

  const {onLogout} = useAuthStateContext();
  const {clearInventories} = useInventoryStateContext();

  const handleLogout = async () => {
    await clearInventories();
    await onLogout();
  };

  return (
    <HomeContainer>
      <FlexRow>
        <AddInventoryBtn
          activeOpacity={0.9}
          onPress={navigateToCreateInventory}>
          <AddInventoryBtnText>Create</AddInventoryBtnText>
        </AddInventoryBtn>
      </FlexRow>
      <InventoryList />
      <LogoutButtonContainer>
        <LogoutBtn
          activeOpacity={0.9}
          onPress={() => {
            setLogoutInventoryModalVisible(true);
          }}>
          <LogoutBtnText>Logout</LogoutBtnText>
        </LogoutBtn>
      </LogoutButtonContainer>
      <ConfirmModal
        isModalVisible={isLogoutInventoryModalVisible}
        title="Are you sure you want to logout?"
        onReject={() => {
          setLogoutInventoryModalVisible(false);
        }}
        onConfirm={handleLogout}
      />
    </HomeContainer>
  );
};

export default Home;
