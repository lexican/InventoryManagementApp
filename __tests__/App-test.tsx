/**
 * @format
 */

import 'react-native';
import React from 'react';

import {act, render, screen} from '@testing-library/react-native';
import Splash from '../src/features/splash/Splash';
import {InventoryContextProvider} from '../src/contexts/inventory-context/InventoryContext';
import Login from '../src/features/login/Login';
import {
  AuthContextProvider,
  AuthStateContext,
} from '../src/contexts/auth-context/AuthContext';
import Toast from 'react-native-toast-message';
import {ModalProps} from 'react-native';
import CreateInventory from '../src/features/create-inventory/CreateInventory';
import Home from '../src/features/home/Home';
import AsyncStorage from '@react-native-async-storage/async-storage';
import InventoryItem from '../src/components/inventory-item/InventoryItem';

jest.mock('react-native-toast-message', () => ({
  show: jest.fn(),
  hide: jest.fn(),
}));

jest.mock('@react-native-async-storage/async-storage', () =>
  require('@react-native-async-storage/async-storage/jest/async-storage-mock'),
);

jest.mock('react-native/Libraries/Modal/Modal', () => {
  const Modal = jest.requireActual('react-native/Libraries/Modal/Modal');
  return (props: ModalProps) => <Modal {...props} />;
});

const createTestProps = (props: Object) => ({
  navigation: {
    navigate: jest.fn(),
    route: jest.fn(),
  },
  ...props,
});

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
    useRoute: () => ({
      params: {},
    }),
  };
});

describe('render screens correctly', () => {
  let props: any;
  beforeEach(() => {
    jest.useFakeTimers();
    props = createTestProps({});
  });

  test('should render splash screen correctly', () => {
    render(<Splash {...props} />);
    expect(screen).toMatchSnapshot();
  });

  it('should render login correctly', () => {
    render(
      <InventoryContextProvider>
        <AuthContextProvider>
          <>
            <Login {...props} />
          </>
        </AuthContextProvider>
      </InventoryContextProvider>,
    );
    expect(screen).toMatchSnapshot();
  });

  it('should render create inventory correctly', () => {
    render(
      <InventoryContextProvider>
        <AuthContextProvider>
          <>
            <CreateInventory {...props} />
          </>
        </AuthContextProvider>
      </InventoryContextProvider>,
    );
    expect(screen).toMatchSnapshot();
  });

  it('should render home correctly', () => {
    render(
      <InventoryContextProvider>
        <AuthContextProvider>
          <Home {...props} />
        </AuthContextProvider>
      </InventoryContextProvider>,
    );
    expect(screen).toMatchSnapshot();
  });
});

describe('edit screen on press test', () => {
  it('navigate to edit inventory on Edit button pressed', async () => {
    render(
      <AuthStateContext.Provider
        value={{
          isLoggedIn: true,
          isLoading: false,
          onLogin: jest.fn(),
          onLogout: jest.fn(),
        }}>
        <InventoryItem
          inventory={{
            name: '',
            price: 300,
            totalStock: 2,
            description: '',
          }}
        />
      </AuthStateContext.Provider>,
    );

    expect(screen).toMatchSnapshot();

    const btn = screen.root.findByProps({testID: 'edit'}).props;

    act(() => btn.onPress());
    expect(btn).toBeTruthy();

    console.log(btn);

    expect(mockedNavigate).toHaveBeenCalled();
  });
});

describe('Async storage CRUD test', () => {
  const mockData = [
    {
      name: 'Banger',
      price: 300,
      totalStock: 2,
      description: 'Not available',
    },
  ];
  beforeEach(() => {
    AsyncStorage.clear();
  });

  it('can read item in async storage', async () => {
    await AsyncStorage.setItem('email', 'test@gmail.com');
    let emailValue = await AsyncStorage.getItem('email');
    expect(emailValue).toBe('test@gmail.com');
  });

  it('can delete item in async storage', async () => {
    await AsyncStorage.setItem('isLoggedIn', 'true');
    await AsyncStorage.removeItem('isLoggedIn');
    let isLoggedInValue = await AsyncStorage.getItem('isLoggedIn');
    expect(isLoggedInValue).toBe(null);
  });

  it('can update item in async storage', async () => {
    await AsyncStorage.setItem('items', JSON.stringify(mockData));

    let items = await AsyncStorage.getItem('items');

    if (items != null) {
      let tempList = JSON.parse(items);
      let tempItem = {...tempList[0]};
      tempItem.name = 'Stock';
      tempList[0] = tempItem;
      await AsyncStorage.setItem('items', JSON.stringify(tempList));
    }

    let newItems = await AsyncStorage.getItem('items');

    if (newItems != null) {
      let items = JSON.parse(newItems);
      expect(items[0].name).toBe('Stock');
    }
  });
});
