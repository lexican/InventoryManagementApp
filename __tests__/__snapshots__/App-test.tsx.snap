// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit screen on press test navigate to edit inventory on Edit button pressed 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ffffff",
        "borderColor": "#f2f4f7",
        "borderRadius": 12,
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "column",
        "height": 110,
        "marginBottom": 12,
        "marginLeft": 20,
        "marginRight": 20,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
        "width": "90%",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#131316",
          "fontSize": 18,
          "fontWeight": "700",
          "marginBottom": 4,
        },
      ]
    }
  />
  <Text
    style={
      [
        {
          "color": "#3a3a44",
          "fontSize": 14,
          "marginBottom": 4,
        },
      ]
    }
  >
    #
    300
  </Text>
  <Text
    style={
      [
        {
          "color": "#3a3a44",
          "fontSize": 14,
        },
      ]
    }
  >
    2
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "height": 40,
        "opacity": 1,
        "position": "absolute",
        "right": 20,
        "top": 20,
        "width": 40,
      }
    }
    testID="edit"
  >
    <Text
      style={
        [
          {
            "color": "#aa77ff",
            "fontSize": 14,
          },
        ]
      }
    >
      Edit
    </Text>
  </View>
</View>
`;

exports[`render screens correctly should render create inventory correctly 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <RCTScrollView
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginBottom": 30,
          "paddingLeft": 20,
          "paddingRight": 20,
        },
      ]
    }
  >
    <View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginBottom": 20,
              "marginTop": 10,
            },
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#aa77ff",
              "borderRadius": 12,
              "height": 40,
              "justifyContent": "center",
              "marginRight": 20,
              "opacity": 1,
              "width": 40,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#ffffff",
                  "fontSize": 24,
                },
              ]
            }
          >
            &lt;
          </Text>
        </View>
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "700",
              },
            ]
          }
        >
          Create
           Inventory
        </Text>
      </View>
      <TextInput
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="Inventory Name"
        secureTextEntry={false}
        style={
          [
            {
              "backgroundColor": "#ffffff",
              "borderColor": "black",
              "borderRadius": 12,
              "borderWidth": 1,
              "height": 44,
              "marginBottom": 12,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
              "width": "100%",
            },
          ]
        }
        value=""
      />
      <TextInput
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="Inventory Total Stock"
        secureTextEntry={false}
        style={
          [
            {
              "backgroundColor": "#ffffff",
              "borderColor": "black",
              "borderRadius": 12,
              "borderWidth": 1,
              "height": 44,
              "marginBottom": 12,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
              "width": "100%",
            },
          ]
        }
        value=""
      />
      <TextInput
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="Inventory Price"
        secureTextEntry={false}
        style={
          [
            {
              "backgroundColor": "#ffffff",
              "borderColor": "black",
              "borderRadius": 12,
              "borderWidth": 1,
              "height": 44,
              "marginBottom": 12,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
              "width": "100%",
            },
          ]
        }
        value=""
      />
      <TextInput
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="Inventory Description"
        secureTextEntry={false}
        style={
          [
            {
              "backgroundColor": "#ffffff",
              "borderColor": "black",
              "borderRadius": 12,
              "borderWidth": 1,
              "height": 44,
              "marginBottom": 12,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
              "width": "100%",
            },
          ]
        }
        value=""
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#aa77ff",
            "borderRadius": 12,
            "marginTop": 20,
            "opacity": 1,
            "paddingBottom": 12,
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 12,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#ffffff",
                "fontSize": 24,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        >
          Create
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`render screens correctly should render home correctly 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "flex-end",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "marginBottom": 10,
          "marginLeft": 0,
          "marginRight": 20,
          "marginTop": 0,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#aa77ff",
          "borderRadius": 12,
          "marginTop": 20,
          "opacity": 1,
          "paddingBottom": 12,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 12,
          "width": 100,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#ffffff",
              "fontSize": 18,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Create
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "height": "60%",
          "justifyContent": "center",
          "paddingLeft": 20,
          "paddingRight": 20,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontSize": 24,
            "textAlign": "center",
          },
        ]
      }
    >
      Inventory is Empty
    </Text>
    <Text
      style={
        [
          {
            "color": "#212121",
            "fontSize": 16,
            "marginTop": 20,
            "textAlign": "center",
          },
        ]
      }
    >
      Add an Inventory by clicking on the create button above
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "marginBottom": 10,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#ffffff",
          "borderColor": "#aa77ff",
          "borderRadius": 12,
          "borderWidth": 1,
          "marginTop": 20,
          "opacity": 1,
          "paddingBottom": 12,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 12,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#aa77ff",
              "fontSize": 24,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Logout
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`render screens correctly should render login correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ffffff",
        "display": "flex",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "paddingLeft": 40,
        "paddingRight": 40,
      },
    ]
  }
>
  <TextInput
    keyboardType="default"
    onChangeText={[Function]}
    placeholder="Email"
    secureTextEntry={false}
    style={
      [
        {
          "backgroundColor": "#ffffff",
          "borderColor": "black",
          "borderRadius": 12,
          "borderWidth": 1,
          "height": 44,
          "marginBottom": 12,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": "100%",
        },
      ]
    }
    value=""
  />
  <TextInput
    keyboardType="default"
    onChangeText={[Function]}
    placeholder="Password"
    secureTextEntry={true}
    style={
      [
        {
          "backgroundColor": "#ffffff",
          "borderColor": "black",
          "borderRadius": 12,
          "borderWidth": 1,
          "height": 44,
          "marginBottom": 12,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": "100%",
        },
      ]
    }
    value=""
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#aa77ff",
        "borderRadius": 12,
        "marginTop": 20,
        "opacity": 1,
        "paddingBottom": 12,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 12,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#ffffff",
            "fontSize": 24,
            "fontWeight": "700",
            "textAlign": "center",
          },
        ]
      }
    >
      Login
    </Text>
  </View>
</View>
`;

exports[`render screens correctly should render splash screen correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#aa77ff",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
          "fontSize": 32,
          "fontWeight": "700",
          "textAlign": "center",
        },
      ]
    }
  >
    Inventory
  </Text>
</View>
`;
